### PR TITLE
tests: Add docker load test

### LINKED ISSUE #290: add docker load test
add integration docker load test

### PR DESCRIPTION
**Pull request by GabyCT**
Add integration docker load test

Fixes #290

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/docker/load_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		Expect(ContainerExists(id)).NotTo(BeTrue())
+		args = []string{"rmi", imageName}
+		runCommand(args, 0)
+	})
+
+	Describe("load with docker", func() {
+		Context("load a container", func() {
+			It("should load image", func() {
+				file, err := ioutil.TempFile(os.TempDir(), "mynewimage.tar")
+				defer os.Remove(file.Name())
```
Comment (devimc) [inline integration/docker/load_test.go]: defer should be used after checking for errors

File: integration/docker/load_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+var _ = Describe("load", func() {
+	var (
+		args      []string
+		id        string
+		exitCode  int
+		imageName string
+		command   *Command
+	)
+
+	runCommand := func(args []string, expectedExitCode int) {
```
Comment (devimc) [inline integration/docker/load_test.go]: this could be a common function

File: integration/docker/load_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	})
+
+	Describe("load with docker", func() {
+		Context("load a container", func() {
+			It("should load image", func() {
+				file, err := ioutil.TempFile(os.TempDir(), "mynewimage.tar")
+				defer os.Remove(file.Name())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(file.Name()).To(BeAnExistingFile())
+				imageName = "test/container-test"
```
Comment (devimc) [inline integration/docker/load_test.go]: could this clash with already existing image names?

File: integration/docker/load_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		Expect(ContainerRemove(id)).To(BeTrue())
+		Expect(ContainerExists(id)).NotTo(BeTrue())
+		args = []string{"rmi", imageName}
+		runCommand(args, 0)
+	})
+
+	Describe("load with docker", func() {
+		Context("load a container", func() {
+			It("should load image", func() {
+				file, err := ioutil.TempFile(os.TempDir(), "mynewimage.tar")
```
Comment (devimc) [inline integration/docker/load_test.go]: you have to close the fd to avoid leaks

File: integration/docker/load_test.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+				defer os.Remove(file.Name())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(file.Name()).To(BeAnExistingFile())
+				imageName = "test/container-test"
+				args = []string{"commit", id, imageName}
+				runCommand(args, 0)
+				args = []string{"save", imageName, "--output", file.Name()}
```
Comment (devimc) [inline integration/docker/load_test.go]: verify that the file is not empty

**Comment by GabyCT** [comment]
@devimc changes applied

File: integration/docker/load_test.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		args = []string{"rmi", imageName}
+		runDockerCommand(0, args...)
+	})
+
+	Describe("load with docker", func() {
+		Context("load a container", func() {
+			It("should load image", func() {
+				file, err := ioutil.TempFile(os.TempDir(), "mynewimage.tar")
+				Expect(err).ToNot(HaveOccurred())
+				defer file.Close()
```
Comment (devimc) [inline integration/docker/load_test.go]: defer is not needed , you can just close the fd here

```
err = file.Close()
Expect(err).ToNot(HaveOccurred())
```

File: integration/docker/load_test.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+				defer os.Remove(file.Name())
+				Expect(file.Name()).To(BeAnExistingFile())
+				imageName = "test/container-test"
+				args = []string{"commit", id, imageName}
+				runDockerCommand(0, args...)
+				args = []string{"save", imageName, "--output", file.Name()}
+				runDockerCommand(0, args...)
+				args = []string{"load", "--input", file.Name()}
+				stdout := runDockerCommand(0, args...)
+				Expect(stdout).To(ContainSubstring(imageName))
```
Comment (devimc) [inline integration/docker/load_test.go]: could you verify that ```file``` is not empty?

Comment (devimc) [inline integration/docker/load_test.go]: dismiss my comment, if save fails load will fail

**Comment by GabyCT** [comment]
@devimc changes applied to the defer, thanks

**Pull request merged**